### ISSUE TITLE
__this.setSize throwing Uncaught Exception

### ISSUE THREAD

**Issue by AtomicAntimatter** [atom/electron#2132]
```
A JavaScript error occured in the browser process
Uncaught Exception:
TypeError: Error processing argument 0.
    at TypeError (native)
    at EventEmitter.<anonymous> (/home/atomic/Downloads/electron-v0.29.1-linux-x64/resources/atom.asar/browser/api/lib/browser-window.js:34:22)
    at emitTwo (events.js:87:13)
    at EventEmitter.emit (events.js:172:7)
```

Going to the code where exception occurred
```
this.webContents.on('move', (function(_this) {
     return function(event, size) {
          return _this.setSize(size);
     };
})(this));
```
Specifically happens at .setSize, my guess is that the function is not implemented or size is an incorrect type. **Commenting out the whole above code solves the issue and produces no visible problems yet.** 

adding Console.log(size) before the return produces [Object object] at the point of error

**Electron Version**: 0.29.1
**OS**: Linux Mint 17.2 Cinnamon 64-bit
**What I was Doing**: Facebook OAuth in a new BrowserWindow instance and grabbing access token from url then returning to main BrowserWindow instance. This issue occurs when resizing the window to a small size. This issue also occurs when progressing naturally through the facebook login dialog. (Facebook probably attempting to resize/query the window size?)

**Comment by hokein** [comment]
You pass the wrong arguments to `BrowserWindow.setSize`.
The callback arguments of `move` event is a dictionary while `setSize` API receives `width` and `height` arguments.

```
{
   'x': <x>,
   'y': <y>,
   'width': <width>,
   'height': <height>,
}
```

So the correct code is
```
this.webContents.on('move', (function(_this) {
     return function(event, rect) {
          return _this.setSize(rect.width, rect.height);
     };
})(this));
```
.

**Comment by AtomicAntimatter** [comment]
This was code written under resources/atom.asar/browser/api/lib/**browser-window.js**
It's part of the electron src, not something I had written.

After modifying to the correct code, the error disappeared AND min-height and min-width were obeyed.
I am not sure how to request this change in the source.

It works very well. Thank you!

**Comment by hokein** [comment]
> This was code written under resources/atom.asar/browser/api/lib/browser-window.js
It's part of the electron src, not something I had written.

Thanks for pointing it out. It's a bug of Electron, fixes in #2138 

BTW, the `move` event of WebContents will be triggerd via calling `window.moveTo` or `window.resizeTo` API and is used internally, developers should not use it since it maybe changed in the future.
